### ISSUE TITLE
FR, ENH: GroupUrlRule treated like UrlManager

### ISSUE THREAD

**Issue by mdmunir** [yiisoft/yii2#12960]
I wish, everything that we can do in `UrlManager` also can be done in `GroupUrlRule`.
#7996 #9122 

```php
'rules' => [
    'post user' => 'user/create',
    [
        'class' => 'yii\web\GroupUrlRule',
        'prefix' => 'api',
        'rules' => [
            'post login' => 'user/login',
            ['class' => 'yii\rest\UrlRule', 'controller' => 'product'],
            [
                'class' => 'yii\web\GroupUrlRule',
                'prefix' => 'v01',
                'rules' => ...
            ],
        ]
    ]
]
```

**Comment by samdark** [comment]
It is better no to create meta-issues. Separate ones are better especially when they already exist.

**Comment by mdmunir** [comment]
if you understand the issue, i am OK. 
I mean, the issue not only how to deal with http method, but how to deal with other type of rule.

**Comment by samdark** [comment]
Yes. The nesting of rules issue was somewhere there. We'll get to it. Or you may try solving it via pull request.

**Comment by rob006** [comment]
I'm going to reopen this. I think that `UrlManager` and `GroupUrlRule` should share the same base. There are multiple features in `UrlManager` that makes perfect sense for `GroupUrlRule`:
* `GroupUrlRule` rule should use its own cache for rules: #14858
* Per-request cache (`UrlManager::getUrlFromCache()`) should work also in `GroupUrlRule` - right now grouping rules may work slower than flat rules structure.
* Settings like `normalizer` or `suffix` could be used also in `GroupUrlRule`,
* Unification of logic may solve some issues (#7996, #9122,  #7296).
